torch/csrc/lazy/core/shape_inference.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <ATen/Tensor.h>
+#include <c10/core/ScalarType.h>
+#include <c10/macros/Export.h>
+#include <c10/util/Optional.h>
+#include <torch/csrc/lazy/core/ir.h>
+#include <torch/csrc/lazy/core/shape.h>
+#include <vector>
+
+namespace torch{
+namespace lazy {
+
+TORCH_API std::vector<Shape> compute_shape__adaptive_avg_pool2d(const at::Tensor & self, at::IntArrayRef output_size);
```

Comment:
Isn't this file codegen'd?

Reply:
In this PR, I changed it to no longer be codegenned.

The rationale was
1) it's confusing to have a handwritten .cpp that is a super-set of functions (shared) between xla/TS backends, but have 2 separate codegenned .h files
1a) worse, it's not straightforward to have the common .cpp file include the backend-specific .h file

2) it's actually _more_ convenient to have the codegen spit out the 'missing' warning at codegen time rather than have it show up at runtime as an undefined symbol, and the codegen can still generate and spit out the entire line you need to add to the header.

3) though i didn't implement it yet, this can be updated to also check and warn/err on inference functions that should be deleted because meta kernels now exist for them.

Reply:
oh I see. yeah that seems reasonable!